lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -763,6 +787,58 @@ void AtenLenTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenAddTensorOp
+//===----------------------------------------------------------------------===//
+
+void AtenAddTensorOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                  MLIRContext *context) {
+  patterns.add(+[](AtenAddTensorOp op, PatternRewriter &rewriter) {
+    // The lhs and rhs of the add.tensor op should be 0d tensors for the
+    // canonicalization to be carried out.
+    // `aten.add.tensor(self, other, alpha)` is canonicalized to
+    // `aten.add.int(self, aten.mul.int(other, alpha))`.
+    Operation *lhsValueTensorLiteralOp =
+        op.self().getDefiningOp<ValueTensorLiteralOp>();
+    Operation *lhsPrimNumToTensorScalarOp =
+        op.self().getDefiningOp<PrimNumToTensorScalarOp>();
+    llvm::Optional<Value> lhs;
+    if (lhsValueTensorLiteralOp) {
+      lhs = getScalarValue(lhsValueTensorLiteralOp, op.getLoc(), rewriter);
+    } else if (lhsPrimNumToTensorScalarOp)
+      lhs = getScalarValue(lhsPrimNumToTensorScalarOp, op.getLoc(), rewriter);
+    else
+      return failure();
+
+    llvm::Optional<Value> rhs;
+    Operation *rhsValueTensorLiteralOp =
+        op.other().getDefiningOp<ValueTensorLiteralOp>();
+    Operation *rhsPrimNumToTensorScalarOp =
+        op.other().getDefiningOp<PrimNumToTensorScalarOp>();
+    if (rhsValueTensorLiteralOp)
+      rhs = getScalarValue(rhsValueTensorLiteralOp, op.getLoc(), rewriter);
+    else if (rhsPrimNumToTensorScalarOp) {
+      rhs = getScalarValue(rhsPrimNumToTensorScalarOp, op.getLoc(), rewriter);
+    } else
+      return failure();
+
+    if (!lhs.hasValue() || !rhs.hasValue())
+      return failure();
+
+    Value mul =
+        rewriter.create<AtenMulIntOp>(op->getLoc(), rhs.getValue(), op.alpha());
+    if (!mul.getType().isa<Torch::IntType>())
```

Comment:
You don't have to check the output type of mul instead the lhs and rhs value passed to mul.